src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -63,7 +97,7 @@ private function completeDefinition($id, Definition $definition)
             return;
         }
 
-        $this->container->addClassResource($reflectionClass);
+        $this->container->addResource(static::createResourceForClass($reflectionClass));
```

Comment:
please wrap this call in a ``isTrackingResources`` call, to avoid creating the resource when unnecessary (which would add a hard dependency to the config component)